Sources/SymbolKit/SymbolGraph/Symbol/FunctionSignature/FunctionParameter.swift

```diff
@@ -14,30 +14,66 @@ extension SymbolGraph.Symbol.FunctionSignature {
     /// An argument of a callable symbol.
     public struct FunctionParameter: Codable {
         enum CodingKeys: String, CodingKey {
-            case name
+            case externalName = "name"
+            case internalName
             case declarationFragments
             case children
         }
 
-        /// The name of the symbol, as referred to in user code (must match the name used in the documentation comment).
-        public var name: String // should we differentiate between internal and external names?
+        /// The name of the function parameter, as referred to in user code (must match the name used in the documentation comment).
```

Comment:
I think that would be confusing for developers using this API. In my opinion the name that developers mean when they talk about a parameters "name" is the internal name. For example, this documentation comment 
```swift
/// Does something
/// - Parameters:
///   - first: Some number
///   - second: Some other number
```
could apply to either this Swift function
```swift
public func doSomething(with first: Int, and second: Int) {}
```
this C function
```c
void doSomething(int first, int second);
```
or this Objective-C method
```objc
- (void)doSomethingWith:(NSInteger)first and:(NSInteger)second;
```

C and Objective-C doesn't have separate external parameter names. In both those cases the `name` are simply "first", and "second".

I started looking into this because I was surprised that for that Swift declaration the function parameter JSON was
```json
{
  "name": "with",
  "internalName": "first",
  "declarationFragments": [
    ...
  ]
}
```
and
```json
{
  "name": "and",
  "internalName": "second",
  "declarationFragments": [
    ...
  ]
}
```

To me it would be more consistent with other languages if it was flipped so that the `"name"` was the internal name and the external name was encoded as "externalName". 

Since the Swift symbol graph generator had already emitted JSON like this for ~4 years I made this change in the SymbolKit decoding instead but if you prefer I can make that change in the Swift symbol graph generator and update the decoding here to look for "name" and "externalName" as the JSON keys.

Regardless of how the JSON looks, I feel like the expected behavior is that `name` always refers to the "internal name" and that in languages that have an explicit external name that's different from the internal name, developers can access that via `externalName`.

Reply:
I could also change this API to only include `name: String` and `externalName: String?` if that's less confusing (and still do switch during JSON decoding)

Reply:
Thinking about this API some more I think that that is the less confusing API to work with.